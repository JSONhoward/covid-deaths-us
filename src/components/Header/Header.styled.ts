import styled from 'styled-components'

export const HeaderContainer = styled('header')`
position: relative;
display: flex;
align-items: center;
height: 4rem;
width: 100%;
color: red;
background-color: white;
box-shadow: 0px 1px 5px rgba(0,0,0,.5);
margin-bottom: .5rem;

h1 {
    text-shadow: 1px 1px 5px rgba(0,0,0,.25);
    margin-left: 2rem;
}

@media screen and (max-width: 400px) {
    font-size: .75rem;
}

@media screen and (max-width: 600px) {
    justify-content: center;
}
`