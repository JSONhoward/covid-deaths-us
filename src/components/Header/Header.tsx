import React from 'react'
import { HeaderContainer } from './Header.styled';

export const HeaderText = 'U.S. Covid-19 Deaths'

const Header = () => {
    return (
        <HeaderContainer>
            <h1>{HeaderText}</h1>
        </HeaderContainer>
    )
}

export default Header
