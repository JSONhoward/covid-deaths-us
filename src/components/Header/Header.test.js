import React from 'react'
import {create} from 'react-test-renderer'
import {screen, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Header from './'
import { HeaderText } from './Header'

describe('Header Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<Header />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Renders Title', () => {
        render(<Header />)

        expect(screen.getByText(HeaderText)).toBeInTheDocument()
    })
})