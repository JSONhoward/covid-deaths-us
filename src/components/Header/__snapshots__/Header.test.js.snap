// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Snapshot Test 1`] = `
<header
  className="sc-bdnylx crJRMx"
>
  <h1>
    U.S. Covid-19 Deaths
  </h1>
</header>
`;
